src/main/java/spoon/Launcher.java

```diff
@@ -420,7 +425,17 @@ protected void processArguments() {
 		environment.setComplianceLevel(jsapActualArgs.getInt("compliance"));
 		environment.setLevel(jsapActualArgs.getString("level"));
 		environment.setAutoImports(jsapActualArgs.getBoolean("imports"));
-		environment.setNoClasspath(jsapActualArgs.getBoolean("noclasspath"));
+
+		if (jsapActualArgs.getBoolean("noclasspath")) {
+			Launcher.LOGGER.warn("The usage of --noclasspath argument is now deprecated: noclasspath is now the default behaviour.");
+			environment.setNoClasspath(true); // avoid getting a warning for boolean not set
+		} else {
+			Launcher.LOGGER.warn("Spoon is now using the 'no classpath mode' by default. If you want to ensure using Spoon in full classpath mode, please use the new flag: --fullclasspath.");
+		}
+
+		if (jsapActualArgs.getBoolean("fullclasspath")) {
```

Comment:
I don't understand why we need a new option which is also boolean.

We can still write `--noclasspath true` or `--noclasspath false`? We only have to change the default value?

Reply:
The argument is a switch, you cannot write `--noclasspath true` or false:
> [urli@vivat ~]$ java -jar spoon-core-6.2.0-jar-with-dependencies.jar --noclasspath true
Error: Unexpected argument: true

So in fact you could only switch on the noclasspath, and never switch off it. The opposite option allows that.

Reply:
I see. Then instead of introducing a closed swith, I would suggest to introduce an open "--mode", "--mode noclasspath", "--mode fullclasspath", that we could extend in the future with new options.

Reply:
Would this mode be related only to classpath? If yes I would prefer to name it `cpmode` to avoid ambiguity.
And do we keep the old switch for backward compatibility or not?

Reply:
By the way, just for information you merge in #2065 a new switch in the usage to disable the comments: it was exactly the same case as this one.